### PR TITLE
MRG: fix tutorial notebook links

### LINKED ISSUE #2604: links to tutorial notebooks are not working!?
on https://sourmash.readthedocs.io/en/latest/tutorials.html#background-and-details, we have four tutorials:

Screenshot 2023-05-02 at 6 09 09 PM

The links to the first three are not working right: they are
```
https://sourmash.readthedocs.io/en/latest/tutorials.html#kmers-and-minhash.md
https://sourmash.readthedocs.io/en/latest/tutorials.html#sourmash-examples.md
https://sourmash.readthedocs.io/en/latest/tutorials.html#sourmash-collections.md
```
the fourth one is correct, but ...it's not a notebook?

https://sourmash.readthedocs.io/en/latest/tutorial-lin-taxonomy.html

So to me it looks like the notebook plugin isn't working right.

### PR DESCRIPTION
**Pull request by ctb**
Fixes https://github.com/sourmash-bio/sourmash/issues/2604

This PR fixes links to binder tutorials that apparently have been broken since 2020 😱 , when the docs were switched over to myst-parser in https://github.com/sourmash-bio/sourmash/pull/1021.

To be fair, it's the kind of subtle bug that would leave users scratching their head going "maybe it's me?" - the links simply went back to the tutorial page... not sure why they weren't flagged by sphinx or myst-parser as being bad!?

This PR also updates environment.yml to properly build:
* includes pip
* specifies minimum python version
* specifies minimum sourmash version

ref https://github.com/sourmash-bio/sourmash/issues/2503 for why specifying the minimum sourmash and/or python versions can be important!

And, finally, I ran and fixed all the notebooks...

Note: I confirmed that this branch launches properly in binder. The URL is: https://mybinder.org/v2/gh/sourmash-bio/sourmash/fix/tutorial-nb

### CODE REVIEW

**Comment by ctb** [comment]
ready for review & merge, @sourmash-bio/devs !

**Comment by ctb** [comment]
bump @ccbaumler @bluegenes

**Comment by ccbaumler** [comment]
The binder do not launch to the notebooks.

**Comment by ctb** [comment]
> The binder do not launch to the notebooks.

thanks! fixed in 0e17b6ed.

File: doc/tutorials.md [hunk 1]
```diff
@@ -13,22 +13,22 @@ X and Linux. They require about 5 GB of disk space and 5 GB of RAM.
 
 ## Background and details
 
-These next four tutorials are all notebooks that you can view, run
+These next three tutorials are all notebooks that you can view, run
 yourself, or run interactively online via the
 [binder](https://mybinder.org) service.
 
-* [An introduction to k-mers for genome comparison and analysis](kmers-and-minhash.md)
+* [An introduction to k-mers for genome comparison and analysis](kmers-and-minhash.ipynb)
```
Comment (ccbaumler) [inline doc/tutorials.md]:
```suggestion
* [An introduction to k-mers for genome comparison and analysis.](kmers-and-minhash.ipynb)
```

File: doc/tutorials.md [hunk 2]
```diff
@@ -13,22 +13,22 @@ X and Linux. They require about 5 GB of disk space and 5 GB of RAM.
...
+* [Working with private collections of signatures.](sourmash-collections.ipynb)
 
-* [Using `sourmash taxonomy` with the LIN taxonomic framework.](tutorial-lin-taxonomy.md)
-
-## More information
+## Advanced tutorials and more information
 
 For more information on analyzing sequencing data with sourmash, check out our [longer tutorial](tutorial-long.md).
 
+Read [using `sourmash taxonomy` with the LIN taxonomic framework.](tutorial-lin-taxonomy.md) for some of our newer taxonomic features.
```
Comment (ccbaumler) [inline doc/tutorials.md]:
```suggestion
Read [using `sourmash taxonomy` with the  Life Identification Number (LIN) taxonomic framework.](tutorial-lin-taxonomy.md) for some of our newer taxonomic features.
```

File: doc/tutorials.md (hunk 1 above)
Comment (ccbaumler) [reply doc/tutorials.md]: The punctuation is non-standard.

**Comment by ctb** [comment]
ok! This should be ready for review again, @ccbaumler - thx :)

**Comment by ctb** [comment]
oh, sorry, didn't see the latest comments! please hold :)

File: doc/tutorials.md (hunk 1 above)
Comment (ctb) [reply doc/tutorials.md]: fixed punctuation in 67399b0330ca6!

**Comment by ctb** [comment]
ready @ccbaumler !

**Comment by ccbaumler** [comment]
LGTM

**Comment by ctb** [comment]
thank you!!

**Pull request merged**